src/Passes/cmake/Library.cmake

```diff
@@ -0,0 +1,38 @@
+macro(list_source_files result  directory)
+  file(GLOB_RECURSE source RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${directory}/*.cpp)
+  set(filelist "")
+  foreach(child ${source})
+    if(NOT ${child} MATCHES "(/Tests/)")
+      list(APPEND filelist ${child})
+    endif()
+  endforeach()  
+
+  set(${result} ${filelist})
+endmacro()
+
+function (microsoft_add_library          
+          library)
+  list(REMOVE_AT ARGV 0)
+
+  set(directory ${CMAKE_CURRENT_SOURCE_DIR}/${library})
+
+  list_source_files(source  ${directory})
+
+  add_library(${library} 
+              SHARED 
+              ${source})
+
+  target_include_directories(
+    ${library} 
+    PRIVATE
+    "${CMAKE_CURRENT_SOURCE_DIR}/include"
+  )
+
+  if(MICROSOFT_ENABLE_DYNAMIC_LOADING)
+    target_link_libraries(${library} 
+      "$<$<PLATFORM_ID:Darwin>:-undefined dynamic_lookup>")
```

Comment:
Is this `PLATFORM_ID:Darwin` something that works on all platforms or is it Mac OS specific?

Reply:
Yes, that is OS X specific. I am not sure whether it is needed for other platforms, but a quick Google suggests that these look substantially different for Linux (and possibly Windows).